docs/src/stories/explorations/Button.stories.jsx

```diff
@@ -0,0 +1,212 @@
+      name: 'leadingVisual',
+      control: {type: 'boolean'},
+      description: 'Slot for SVG icon or emoji (boolean only for testing purposes)',
+      defaultValue: false,
+      table: {
+        category: 'Slot'
+      }
+    },
+    trailingVisual: {
+      name: 'trailingVisual',
+      control: {type: 'boolean'},
+      description: 'Slot for SVG icon or emoji (boolean only for testing purposes)',
+      table: {
+        category: 'Slot'
+      },
+      defaultValue: false
+    },
+    trailingAction: {
+      defaultValue: false,
+      control: {type: 'boolean'},
+      description:
+        'Slot for SVG icon that indicates an action. Primarily used by other Primer components, like a DropdownMenu or overlay trigger (boolean only for testing purposes)',
+      table: {
+        category: 'Slot'
+      }
+    },
+    pressed: {
+      defaultValue: false,
+      control: {type: 'boolean'},
+      table: {
+        category: 'State'
+      }
+    },
+    focusElement: {
+      control: {type: 'boolean'},
+      description: 'set focus on one element',
+      table: {
+        category: 'State'
+      }
+    },
+    active: {
+      control: {type: 'boolean'},
+      description: 'set button to active state',
+      table: {
+        category: 'State'
+      }
+    }
+  }
+}
+
+const focusMethod = function getFocus() {
+  // find the focusable element
+  var button = document.getElementsByTagName('button')[0]
+  // set focus on element
+  button.focus()
+}
+
+const star = (
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16">
+    <path
+      fill-rule="evenodd"
+      d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25z"
+    ></path>
+  </svg>
+)
+
+const caret = (
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16">
+    <path d="M4.427 7.427l3.396 3.396a.25.25 0 00.354 0l3.396-3.396A.25.25 0 0011.396 7H4.604a.25.25 0 00-.177.427z"></path>
+  </svg>
+)
+
+export const ButtonTemplate = ({
+  label,
+  variant,
+  disabled,
+  size,
+  fullWidth,
+  leadingVisual,
+  trailingVisual,
+  trailingAction,
+  pressed,
+  focusElement,
+  active,
+  visualPosition,
+  className,
+  ariaLabel
+}) => (
+  <>
+    <button
+      disabled={disabled}
+      className={clsx(
+        'Button',
+        className && `${className}`,
+        variant && `${variant}`,
+        size && `${size}`,
+        fullWidth && 'Button--fullWidth',
+        active && 'Button--active'
+      )}
+      aria-pressed={pressed ? pressed : undefined}
+      aria-label={ariaLabel ? ariaLabel : undefined}
+    >
+      {/* {leadingVisual && <span className="" dangerouslySetInnerHTML={{__html: leadingVisual}} />} */}
+      <span className={clsx(visualPosition && `${visualPosition}`, 'Button-content')}>
```

Comment:
@pksjce I made some grid and markup adjustments to Button to be able to support "locked" icon positioning. This basically takes `trailingAction` out of the grid, as we've decided it will always be locked right even if other icons are not locked to the container. I think its too early in the design process to do a thorough code review but if you have any reservations here let me know! 

I always try and keep as many divs/spans out of a component as I can.. but I couldn't find a nicer way to both lock one icon to the right while centering the rest of the button contents.